examples/fit_ESR_voigt.py

```diff
@@ -0,0 +1,137 @@
+from pint import UnitRegistry
+from pyspecdata import *
+from pyspecProcScripts import *
+from pyspecProcScripts import QESR_scalefactor
+from sympy import symbols, Symbol, latex
+from sympy import exp as s_exp
+import numpy as np
+
+logger = init_logging(level="debug")
+fieldaxis = "$B_0$"
+exp_type = "francklab_esr/alex"
+with figlist_var() as fl:
+    for filenum, (thisfile, calibration, diameter, background) in enumerate(
+        [
+            (
+                "230418_200uM_TEMPOL_3_redo2.DSC",
+                "230202",
+                "QESR caps",
+                find_file("230418_water.DSC", exp_type=exp_type)["harmonic", 0],
+            ),
+        ]
+    ):
+        # {{{ load data in and rescale and center about 0
+        d = find_file(thisfile, exp_type=exp_type)
+        if "harmonic" in d.dimlabels:
+            d = d["harmonic", 0]
+        color = d.get_plot_color()
+        d /= QESR_scalefactor(d, calibration_name=calibration, diameter_name=diameter)
+        center_field = d.getaxis(fieldaxis)[r_[0, -1]].mean()
+        d -= d[fieldaxis, -100:].data.mean()
+        d.setaxis(fieldaxis, lambda x: x - center_field)
+        background = ndshape(d).alloc()
+        background.copy_axes(d)
+        background = background.interp(fieldaxis, d.getaxis(fieldaxis))
+        d -= background
+        # }}}
+        d.ift(fieldaxis, shift=True)
+        d *= 2
+        d[fieldaxis:0] *= 0.5
+        d.ft(fieldaxis)
+        # {{{ Plot starting spectra
+        d.set_units(fieldaxis, "G")
+        fl.next("B domain")
+        fl.plot(d, label="actual data")
+        d.ift(fieldaxis)
+        fl.next("u domain")
+        fl.plot(d, label="real data")
+        fl.plot(d.imag, label="imag data")
+        # }}}
+        # {{{ Make list of guesses
+        centers = [
+            -21,
+            -5,
+            13,
+        ]  # make a list of the centers of the starting data for the fits
+        lambda_l_guess = [1.2, 1.2, 1.2]
+        lambda_g_guess = [1.2, 1.2, 1.2]
+        amp_guess = [120, 120, 120]
+        # }}}
+        # {{{Initiate lmfit
+        d.rename(fieldaxis, "u")
+        u = d.fromaxis("u")
+        f = lmfitdata(d)
+        # }}}
+        # {{{ make symbols and prep lists for symbols
+        list_of_lambda_ls = [
+            symbols("lambda_l%d" % (j + 1), real=True) for j in range(3)
+        ]
+        list_of_lambda_gs = [
+            symbols("lambda_g%d" % (j + 1), real=True) for j in range(3)
+        ]
+        list_of_amps = [symbols("amp%d" % (j + 1), real=True) for j in range(3)]
+        list_of_Bc = [symbols("B%d" % (j + 1), real=True) for j in range(3)]
```

Comment:
each of these should be nested in a dictionary comprehension, something like this:
```
all_symb = {symbname:[symbols("%s%d" % (symbname,(j+1)), real=True) for j in range(3)] for symbname in ["lambda_l", "lambda_g", "amp", "B"]}
```